core/consensus/grandpa/impl/grandpa_impl.cpp

```diff
@@ -538,17 +540,25 @@ namespace kagome::consensus::grandpa {
       logger_->warn("Vote signed by unknown validator");
       return;
     };
+
+    bool isPrevotesChanged = false;
+    bool isPrecommitsChanged = false;
     visit_in_place(
         msg.vote.message,
-        [&target_round, &msg](const PrimaryPropose &) {
-          target_round->onProposal(msg.vote);
+        [&](const PrimaryPropose &) {
+          target_round->onProposal(msg.vote, true);
```

Comment:
I think you could just call sendProposal here. And not change interface.

Reply:
It would, but send* methods are private